screens/BooksScreen.js

```diff
@@ -21,28 +21,69 @@ export default function BooksScreen() {
     getBooks();
   }, []);
 
-  const authorName = data.map((e) => {
-    e.authors.map((author) => {
-      console.log(author.name)
-      return author.name
-    })
-     });
+  //looking for ISBN
+  const availabilities =  data.map((book) => {
+    return !!book && book.availability
+  });
+
+  const isbnArray = availabilities.map((availability)=>{
+    console.log("availability.isbn:", availability !== undefined ? availability.isbn : '')
+    return availability !== undefined ? availability.isbn : ''
+  })
+    console.log("isbnArray:", isbnArray)
+
+  const isbn = isbnArray.forEach((isbn) => {
+    console.log("ISBN:",  typeof isbn === "string" ? isbn : '')
+    return typeof isbn === "string" ? `[ISBN: ${isbn}]` : ''
+  })
+  console.log("constISBN:", isbn)
+
+  const renderItem = ({ item }) => {
+
+    const authors = item.authors.length > 1
+      ? `[authors: ${item.authors[0].name} and ${item.authors.length - 1} more]`
+      : `[author: ${item.authors[0].name}]`
+      // console.log(item.authors.length)
+
+    return (
+      <View>
+        <Text style={styles.item}>
+          {item.title} {authors} {isbn}
+        </Text>
+
+        {/* {isbnArray.forEach(isbn =>
+          <Text>
+            {typeof isbn === "string" ? isbn : '' }
+          </Text>
+        )} */}
+
+      </View>
+    );
+  };
 
   return (
     <View style={{ flex: 1, padding: 24 }}>
-      {isLoading ? <ActivityIndicator /> : (
 
-      <FlatList
+      {isLoading ? <ActivityIndicator /> : (
+        <FlatList
           data={data}
           key={({ id }, index) => id}
```

Comment:
this is not giving me issues or warnings . Cannot find the example where I copied it from, but in RN docu for flatlist it's also a function https://reactnative.dev/docs/flatlist

Reply:
I can not find a `key` property of the `FlatList` in the documentation you referred to.

Reply:
from code example and here [keyextractor](https://reactnative.dev/docs/flatlist#keyextractor) is "explained". To be honest I understand just the first sentence there..
<img width="820" alt="Screenshot 2021-11-02 at 14 34 41" src="https://user-images.githubusercontent.com/43641321/139857116-037b4e6a-5bd0-4ce5-bca7-5caab5c74909.png">

Reply:
there is no `key` prop in your example.

Reply:
I guess key `property` and `keyExtractor` are not the same thought .

Reply:
I do not know why the warning goes away in the case of you adding a function as the `key` prop of the `FlatList`.
do you maybe get another error or warning instead?

The reason why the warning shows up is not because of the `key` of the `FlatList` but because of its multiple children!

It is not surprising that the warning does not go away, when you do `keyExtractor={ item => item.id}` if the items do not have an `id`, is it? Instead of not defining a `key` for the children, you actively give them. `undefined` instead. ;)

The `key` needs to be a unique value for each of the children. How that value is constructed is up to you. There is no magic answer provided by react/react-native.

Using a `key` which only depends on the `index` can cause issues, as explained in the react documentation, but until you run into those specific issues, it works.

Reply:
Not getting other error or warning.  But I have a felling I might have to deal with this when I work on this step: "When reaching the end of the list, fetch more entries of the chosen subject and add them to the list."